pkg/apis/serving/v1beta1/configmap.go

```diff
@@ -280,7 +280,7 @@ func NewServiceConfig(clientset kubernetes.Interface) (*ServiceConfig, error) {
 	if err != nil {
 		return nil, err
 	}
-	serviceConfig := &ServiceConfig{}
+	serviceConfig := &ServiceConfig{ServiceClusterIPNone: true}
```

Comment:
Any particular reason for doing this here rather than in the [configmap-patch](https://github.com/opendatahub-io/kserve/blob/release-v0.14/config/overlays/odh/inferenceservice-config-patch.yaml)?

Reply:
[CPD docs ](https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2.8/html/installing_and_uninstalling_openshift_ai_self-managed/preparing-openshift-ai-for-ibm-cpd_prepare-openshift-ai-ibm-cpd#editing-model-inferencing-configuration-ibm-cpd_prepare-openshift-ai-ibm-cpd)tell users to use the `opendatahub.io/managed=false` to change some other configs. Patching the manifest will be OK for new installs, but upgrades would still see the issue.

This is why we had to change the programmatic default.